clientd3d/mapfile.c

```diff
@@ -101,31 +116,51 @@ bool MapFileVerifyHeader(FILE *file, bool file_exists)
 
    if (file_exists)
    {
-     for (i=0; i < 4; i++)
-       if (fread(&byte, 1, 1, file) != 1 || byte != map_magic[i])
-         return false;
-     if (fread(&version, 1, 4, file) != 4 || version > MAPFILE_VERSION)
-     {
-       debug(("Bad map file version %d; expecting %d\n", version, MAPFILE_VERSION));
-       return false;
-     }
+      for (i = 0; i < 4; i++)
+         if (fread(&byte, 1, 1, file) != 1 || byte != map_magic[i])
+            return false;
+      if (fread(&version, 1, 4, file) != 4)
+         return false;
+      if(version == MAPFILE_VERSION_1)
+      {
+         // Migrate old version 1 map file to version 2
+         debug(("Migrating map file from version 1 to version 2\n"));
+         if (!MapFileMigrateVersion1ToVersion2())
+         {
+            debug(("Map file migration failed!\n"));
+            return false;
+         }
+      }
+      else if (version != MAPFILE_VERSION)
+      {
+         debug(("Bad map file version %d; expecting %d\n", version, MAPFILE_VERSION));
+         return false;
```

Comment:
We may want to consider what happens when we remove the migration code some months from now.  If someone *does* log in with a v1 file after we've removed the migration code, by default this will result in a permanently broken map.  Should we delete the file and start a new one in that case?